src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Server/ServerHandler.cs

```diff
@@ -91,5 +92,23 @@ private int GetServerMinMemory()
         {
             return configService.GetServerSmoConfig(server, configService.MinServerMemoryPropertyNumber).ConfigValue;
         }
+
+        private void UpdateServerProperties(InitializeViewRequestParams viewParams, ServerInfo serverObj)
+        {
+            if (viewParams != null)
+            {
+                ConnectionInfo connInfo = this.GetConnectionInfo(viewParams.ConnectionUri);
+                ServerConnection serverConnection = ConnectionService.OpenServerConnection(connInfo, ObjectManagementService.ApplicationName);
+                this.server = new Server(serverConnection);
```

Comment:
Missed this earlier - we shouldn't be storing state like this. This is a singleton instance for handling ALL requests so the server will change from request to request. And while it's unlikely to have two overlapping given how the UI works it's still something we should write this code in a functional way.

Look at how the other handlers are doing this for an example